stagemonitor-core/src/main/java/org/stagemonitor/core/elasticsearch/ElasticsearchClient.java

```diff
@@ -115,15 +127,36 @@ public int sendAsJson(final String method, final String path, final Object reque
 				.build());
 	}
 
-	public void updateKibanaIndexPattern(final String indexPatternLocation, final String elasticsearchKibanaIndexPatternPath) {
+	public void updateKibanaIndexPattern(final String indexName, final String indexPatternLocation) {
+		updateKibanaIndexPattern(indexName, indexPatternLocation, false);
+	}
+
+	public void updateKibanaIndexPattern(final String indexName, final String indexPatternLocation, boolean setAsDefaultIndex) {
+		final String elasticsearchKibanaIndexPatternPath = isElasticsearch6() ? "/.kibana/doc/index-pattern:" + indexName : "/.kibana/index-pattern/" + indexName;
 		logger.debug("Sending index pattern {} to {}", indexPatternLocation, elasticsearchKibanaIndexPatternPath);
 		try {
 			ObjectNode stagemonitorPattern = JsonUtils.getMapper().readTree(IOUtils.getResourceAsStream(indexPatternLocation)).deepCopy();
-			stagemonitorPattern.put("fields", getFields(stagemonitorPattern.get("fields").asText()));
+
+			if (isElasticsearch6()) {
+				ObjectNode indexPatternNode = (ObjectNode) stagemonitorPattern.get("index-pattern");
+				indexPatternNode.put("fields", getFields(stagemonitorPattern.get("index-pattern").get("fields").asText()));
+			} else {
+				stagemonitorPattern.put("fields", getFields(stagemonitorPattern.get("fields").asText()));
+			}
+
 			JsonNode currentPattern = fetchCurrentKibanaIndexPatternConfiguration(elasticsearchKibanaIndexPatternPath);
 			JsonNode mergedDefinition = JsonMerger.merge(currentPattern, stagemonitorPattern,
 					mergeStrategy().mergeEncodedObjects("fieldFormatMap").encodedArrayWithKey("fields", "name"));
+
 			sendAsJson("PUT", elasticsearchKibanaIndexPatternPath, mergedDefinition);
+			if (setAsDefaultIndex) {
```

Comment:
I like the idea! But what if the user already has configured a default index in this Kibana instance, for example for logstash?

Reply:
I could check if a default index is set already. I think this change should be done in a separate commit and removed it from this one.